tools/extract_dependencies.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+
+# Author: Jonpas
+
+import os
+import sys
+import re
+import shutil
+
+######## GLOBALS #########
+FILE_EXTRACTED = "temp\\dependencies.temp"
+##########################
+
+def writeToFile(string):
+    with open(FILE_EXTRACTED, "a") as file:
+        file.write("{}\n".format(string))
+
+
+def main():
+    print("""
+    ####################################
+    # Extract ACE3 Module Dependencies #
+    ####################################
+    """)
+
+    scriptpath = os.path.realpath(__file__)
+    projectpath = os.path.dirname(os.path.dirname(scriptpath))
+    addonspath = os.path.join(projectpath, "addons")
+
+    # Prepare directory and file
+    if not os.path.exists("temp"):
+        os.makedirs("temp")
+
+    open(FILE_EXTRACTED, "w").close()
+
+    # Iterate through folders in the addons directory
+    for folder in next(os.walk(addonspath))[1]:
+        writeToFile(folder)
+        print("Module: {}".format(folder))
+
+        # Open config.cpp file
+        with open(os.path.join(addonspath, folder, "config.cpp")) as file:
+            for line in file:
+                # Extract dependencies
+                match = re.match(r"\s+requiredAddons\[]\ = {(.+)};", line)
```

Comment:
Yeah, this is one of those situations where it'll be fine if all our addons use a one-line format. However a parser of some description is probably more suitable